cronsetup/cronsetup.go

```diff
@@ -0,0 +1,62 @@
+package cronsetup
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Scalingo/go-etcd-cron"
+	"github.com/Scalingo/go-utils/logger"
+	"github.com/coreos/etcd/clientv3"
+)
+
+type SetupOpts struct {
+	EtcdConfig func() (clientv3.Config, error)
+	Jobs       []etcdcron.Job
+}
+
+func Setup(ctx context.Context, opts SetupOpts) (func(), error) {
+	etcdConfig, err := opts.EtcdConfig()
+	if err != nil {
+		return nil, fmt.Errorf("fail to get etcd v3 config: %v", err)
+	}
+
+	etcdMutexBuilder, err := etcdcron.NewEtcdMutexBuilder(etcdConfig)
+	if err != nil {
+		return nil, fmt.Errorf("fail to get etcd mutex builder: %v", err)
+	}
+
+	funcCtx := func(ctx context.Context, j etcdcron.Job) context.Context {
```

Comment:
Is it really what you wanted to do? A function which just do logging?

Reply:
What else would you like to do?